README.md

```diff
@@ -1,2 +1,422 @@
+public struct ParentView: View {
+    public var body: some View {
+        VStack {
+            Text("Child View")
+            childViewBuilder.instantiate()
+        }
+    }
+
+    public init(childViewBuilder: Instantiator<some View>) {
+        self.childViewBuilder = childViewBuilder
+    }
+
+    // The Instantiator‘s `instantiate()` method will build a view of type `ChildView`.
+    // Because the type is passed in as a string literal, this code does not need to
+    // have a dependency on the module that defines `ChildView`. All that is required
+    // for this code to compile is for there to be an
+    // `@Instantiable public struct ChildView: View` in the codebase.
+    @Instantiated(fulfilledByType: "ChildView")
+    private let childViewBuilder: Instantiator<some View>
+}
+```
+
+### @Forwarded
+
+Property declarations within [`@Instantiable`](#instantiable) types decorated with the [`@Forwarded` macro](Sources/SafeDI/PropertyDecoration/Forwarded.swift) are forwarded into the SafeDI dependency tree by a [`ForwardingInstantiator`](Sources/SafeDI/DelayedInstantiation/ForwardingInstantiator.swift) instance’s `instantiate(…)` method. A `@Forwarded`-decorated property is available to be [`@Received`](#received) by objects instantiated further down the dependency tree.
+
+A single `@Instantiable` type may have at most one `@Forwarded`-decorated property.
+
+### @Received
+
+Property declarations within [`@Instantiable`](#instantiable) types decorated with the [`@Received` macro](Sources/SafeDI/PropertyDecoration/Received.swift) are injected into the enclosing type‘s initializer. Received properties must be [`@Instantiated`](#instantiated) or [`@Forwarded`](#forwarded) by an object higher up in the dependency tree.
+
+### Delayed instantiation
+
+When you want to instantiate a dependency after your `init(…)`, you need to declare an `Instantiator<Dependency>`-typed property as `@Instantiated` or `@Received`.
+
+#### Instantiator
+
+The `Instantiator` type is how SafeDI enables deferred instantiation of an `@Instantiable` type. `Instantiator` has a single generic that matches the type of the to-be-instantiated instance. Creating an `Instantiator` property is as simple as creating any other property in the SafeDI ecosystem:
+
+```swift
+@Instantiable
+public struct MyApp: App {
+    public var body: some Scene {
+        WindowGroup {
+            // Returns a new instance of a `ContentView`.
+            contentViewInstantiator.instantiate()
+        }
+    }
+
+    public init(contentViewInstantiator: Instantiator<ContentView>) {
+        self.contentViewInstantiator = contentViewInstantiator
+    }
+
+    /// A private property that knows how to instantiate a content view.
+    @Instantiated
+    private let contentViewInstantiator: Instantiator<ContentView>
+}
+```
+
+It is possible to write a `Instantiator` with a type-erased generic by utilizing `@Instantiated`‘s `fulfilledByType` parameter.
+
+#### ForwardingInstantiator
```

Comment:
Yes! If you have a `@Forwarded` property you _must_ use a `ForwardingInstantiator` to instantiate the thing. Sounds like I need to make that more clear.

Reply:
Could Forwarded be a property wrapper (instead of a macro) that automatically wraps the forwarded type in a `ForwardingInstantiator`?

Reply:
sorry, the `ForwardingInstantiator` is how you instantiate the _enclosing type_ that has a forwarded property, not the forwarded property itself. No way to really make this magical, especially because the `ForwardingInstantiator` needs the `@Forwarded` property as an argument in its `instantiate(...)` method.

Reply:
from what I remember, the macro is only used as a syntax flag for the plugin, so making this a property wrapper should work.

Reply:
if i'm understanding this right

Reply:
oh yeah I've played with utilizing property wrappers instead of macros and it does work (I did this – and then deleted – for `@LazyInstantiated`).

Are you saying you would have something like
```
@Instantiated // macro
@Forwarded<ForwardArgument> // property wrapper
let builder: Instantiator<InstantiatedType>
```

Because we can't make `@Forwarded` itself a property wrapper that exposes a `ForwardingInstantiator` because the forwarding instantiator is the thing we need to _create the object with the forwarded property_.

All that said: what problem are we solving here?

Reply:
okay I misunderstood—I thought you were saying this needed to be:
```
let builder: FordwardingInstantiator<InstantiatedType>
```

But considering that it is not, property wrappers don't make sense here. 

I was less trying to solve a problem, and more trying to understand how this works.

Reply:
Got it. We need:
```
let builder: FordwardingInstantiator<ForwardedArgument, InstantiatedType>
```

The code example a few lines down hopefully makes this a bit more clear... but I'm very much hearing I need more English explanation of this requirement too.

Reply:
I updated the `@Forwarded` description to make this a bit more clear.